Core/clim-core/menu.lisp

```diff
@@ -282,8 +282,8 @@ account, and create a list of menu buttons."
 
 ;;; Menu creation from command tables
 
-(defparameter *enabled-text-style*  (make-text-style :sans-serif :roman :normal))
-(defparameter *disabled-text-style* (make-text-style :sans-serif :roman :normal))
+(defparameter *enabled-text-style*  (make-text-style :sans-serif nil nil))
+(defparameter *disabled-text-style* (make-text-style :sans-serif nil nil))
```

Comment:
since we have to create a text-style on each invocation *anyway* I think that we should inline them (and remove these special variables). Each of them is used only twice in the same function. We won't need merging text style then as well, because (make-text-style …) fills NIL's with the default one.

Reply:
My reasoning might be wrong, but I think the underlying issue is that while labels and other text associated with gadgets is output in the fully specified style eventually produced by medium-merged-text-style, the internal layout of gadgets (and compose-space etc) is guided by pane-text-style, which might have unspecified components. When the size is unspecified, methods such as text-style-ascent seem to take it as 12. To avoid this mismatch, my commits were intended to fully specify the pane text style from the outset. Another solution could be to simulate the outcome of medium-merged-text-style throughout gadgets.lisp - but that would require very many changes.

Reply:
Thank you for the explanation, I'll look into the code to have a better understanding of that.

Reply:
OK, so the problem originated in extensions/fonts module. Please see branch https://github.com/McCLIM/McCLIM/tree/default-text-style-fixes for a commit which fixes issue #624. It also modifies text-style initarg to be put directly in basic-gadget (only once) -- second commit. Please rebase your remaining changes (compose-space fixes and suggested-height against that branch).

I'm not doing that myself because that would be kind of hijack of the pull request. This fix is a direct result of peer reviewing your work, so all credit belongs to you.

Reply:
Thanks for finding and fixing the problem! I don't need any credit, and I'm getting myself thoroughly confused in trying to rebase the remaining changes - probably due to the fact that I'm going down with flu. So please go ahead and make those changes yourself (as in commit https://github.com/McCLIM/McCLIM/pull/655/commits/1ace64b6facef6fe2ec50ed9f1412a2ed3820aba and changing :suggest- -> :suggested- x2)